script.js

```diff
@@ -0,0 +1,346 @@
+  document.querySelector('.mdl-fullscreen .mdl-text p .span5').textContent = worksArray[j].language2;
+  document.querySelector('.mdl-fullscreen .mdl-text p .span6').textContent = worksArray[j].language3;
+  document.querySelector('.mdl-fullscreen .mdl-text p .span7').textContent = worksArray[j].language4;
+  document.querySelector('.mdl-fullscreen .mdl-text p .span8').textContent = worksArray[j].language5;
+  document.querySelector('.mdl-fullscreen .mdl-text p .span9').textContent = worksArray[j].language6;
+  document.querySelector('.wrapper .wrapper-btns .btnA').textContent = 'See Live';
+  document.querySelector('.wrapper .wrapper-btns .btnB').textContent = 'See Source';
+  document.querySelector('.mdl-fullscreen .mdl-text .btnA').textContent = 'See Live';
+  document.querySelector('.mdl-fullscreen .mdl-text .btnB').textContent = 'See Source';
+  divSpan1.addEventListener('click', () => {
+    popupDiv.remove();
+  });
+  seeLiveBtn.addEventListener('click', () => {
+    window.open(worksArray[j].liveBtn);
+  });
+  seeSourceBtn.addEventListener('click', () => {
+    window.open(worksArray[j].sourceBtn);
+  });
+}
+
+btn1.addEventListener('click', () => {
+  j = 0;
+  popup();
+});
+
+btn2.addEventListener('click', () => {
+  j = 1;
+  popup();
+});
+
+btn3.addEventListener('click', () => {
+  j = 2;
+  popup();
+});
+
+btn4.addEventListener('click', () => {
+  j = 3;
+  popup();
+});
+
+function showMessage(input, message, type) {
+  const msg = input.parentNode.querySelector('small');
+  msg.innerText = message;
+  input.className = type ? 'success' : 'error';
+  return type;
+}
+function showError(input, message) {
+  return showMessage(input, message, false);
+}
+
+function showSuccess(input) {
+  return showMessage(input, '', true);
+}
+
+function hasValue(input, message) {
+  if (input.value.trim() === '') {
+    return showError(input, message);
+  }
+  return showSuccess(input);
+}
+
+function validateEmail(input, requiredMsg, invalidMsg) {
+  // check if the value is not empty
+  if (!hasValue(input, requiredMsg)) {
+    return false;
+  }
+  const emailRegex = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+
+  const email = input.value.trim();
+  if (!emailRegex.test(email)) {
+    return showError(input, invalidMsg);
+  }
+  return true;
+}
+
+const form = document.querySelector('#signup');
+
+const NAME_REQUIRED = 'Please enter your name';
+const EMAIL_REQUIRED = 'Please enter your email';
+const EMAIL_INVALID = 'Please enter a correct email address format';
+
+form.addEventListener('submit', (event) => {
+  event.preventDefault();
+  const nameValid = hasValue(form.elements.name, NAME_REQUIRED);
+  const emailValid = validateEmail(form.elements.email, EMAIL_REQUIRED, EMAIL_INVALID);
+
+  if (nameValid && emailValid) {
+    // eslint-disable-next-line no-alert
+    alert('No form was posted!');
+  }
+});
```

Comment:
Great job @Oscah01 🤝

- [ ] Kindly make sure your form submits if the validation passes (lowercase email). As for your case, both lowercase and uppercase emails are failing the validation.
- [ ] According to [Javascript best practices](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/javascript_best_practices.md) it is not advisable to use `alert ()`. This is because it is not easy to style.
- [ ] Kindly make sure you display the error message as stated in the [project requirements](https://github.com/microverseinc/curriculum-html-css/blob/main/portfolio/9_validate_form.md).
- Your form validation
![alert](https://user-images.githubusercontent.com/63917725/181782044-6bbc4e53-c8ab-4513-9f25-1c3f07ca1351.PNG)